onnxruntime/core/providers/nnapi/nnapi_builtin/model.h

```diff
@@ -133,11 +133,49 @@ class Model {
 
   void SetShaper(const Shaper shaper) { shaper_ = shaper; }
 
-  void SetInputBuffer(const int32_t index, const InputBuffer& input);
-  void SetOutputBuffer(const int32_t index, const OutputBuffer& output);
+  int32_t GetAndroidSdkVer() const;
+};
+
+class Execution {
+  friend class Model;
+
```

Comment:
Wondering if this is necessary? Do we really need a private ctor and a friend relationship to stop someone else creating an instance of this class?

Reply:
You are right, this is not necessary, will keep this for no, and change in my next PR, to avoid re-run all the checks and need sign-off again

---
In reply to: [462749575](https://github.com/microsoft/onnxruntime/pull/4650#discussion_r462749575) [](ancestors = 462749575)

Reply:
Resolve conflict with master cleared the sign-off status, fixed in new iteration.:disappointed: 

---
In reply to: [463201299](https://github.com/microsoft/onnxruntime/pull/4650#discussion_r463201299) [](ancestors = 463201299,462749575)